third_party/move/move-vm/types/src/values/values_impl.rs

```diff
@@ -410,12 +417,13 @@ impl ValueImpl {
             Bool(x) => Bool(*x),
             Address(x) => Address(*x),
 
-            ContainerRef(r) => ContainerRef(r.copy_value()),
-            IndexedRef(r) => IndexedRef(r.copy_value()),
+            // Note: refs copy only clones Rc, so no need to increment depth.
+            ContainerRef(r) => ContainerRef(r.copy_by_ref()),
+            IndexedRef(r) => IndexedRef(r.copy_by_ref()),
 
             // When cloning a container, we need to make sure we make a deep
             // copy of the data instead of a shallow copy of the Rc.
-            Container(c) => Container(c.copy_value()?),
+            Container(c) => Container(c.copy_value(depth, max_depth)?),
```

Comment:
Document why `depth` does not have to be incremented here.

Reply:
done